api/src/importers/nro/base-record-utils.js

```diff
@@ -0,0 +1,191 @@
+ *
+ * @class BaseRecordUtils
+ */
+class BaseRecordUtils {
+  /**
+   * Creates an instance of BaseRecordUtils.
+   *
+   * @param {*} auth_payload user information for auditing
+   * @param {*} recordType an item from record-type-enum.js -> RECORD_TYPE
+   * @param {*} csvRow an array containing the values from a single csv row.
+   * @memberof BaseRecordUtils
+   */
+  constructor(auth_payload, recordType, csvRow) {
+    if (!recordType) {
+      throw Error('BaseRecordUtils - required recordType must be non-null.');
+    }
+
+    this.auth_payload = auth_payload;
+    this.recordType = recordType;
+    this.csvRow = csvRow;
+  }
+
+  /**
+   * Transform an nro-csv row into a NRPTI record.
+   *
+   * Note: Only transforms common fields found in ALL supported nro-csv types.
+   *       To include other values, extend this class and adjust the object returned by this function as needed.
+   *
+   * @param {object} csvRow nro-csv row (required)
+   * @returns {object} NRPTI record.
+   * @throws {Error} if record is not provided.
+   * @memberof BaseRecordUtils
+   */
+  transformRecord(csvRow) {
+    if (!csvRow) {
+      throw Error('transformRecord - required csvRow must be non-null.');
+    }
+
+    return {
+      _schemaName: this.recordType._schemaName,
+      recordType: this.recordType.displayName,
+
+      sourceSystemRef: 'nro-inspections-csv'
+    };
+  }
+
+  /**
+   * Searches for an existing master record, and returns it if found.
+   *
+   * @param {*} nrptiRecord
+   * @returns {object} existing NRPTI master record, or null if none found or _sourceRefNrisId is null
+   * @memberof BaseRecordUtils
+   */
+  async findExistingRecord(nrptiRecord) {
+    if (!nrptiRecord._sourceRefNrisId) {
+      return null;
+    }
+
+    const masterRecordModel = mongoose.model(this.recordType._schemaName);
+
+    return await masterRecordModel
+      .findOne({
+        _schemaName: this.recordType._schemaName,
+        _sourceRefNrisId: nrptiRecord._sourceRefNrisId
+      })
+      .populate('_flavourRecords', '_id _schemaName');
+  }
+
+  /**
+   * Update an existing NRPTI master record and its flavour records (if any).
+   *
+   * @param {*} nrptiRecord
+   * @param {*} existingRecord
+   * @returns {object} object containing the update master and flavour records (if any)
+   * @memberof BaseRecordUtils
+   */
+  async updateRecord(nrptiRecord, existingRecord) {
+    if (!nrptiRecord) {
+      throw Error('updateRecord - required nrptiRecord must be non-null.');
+    }
+
+    if (!existingRecord) {
+      throw Error('updateRecord - required existingRecord must be non-null.');
+    }
+
+    try {
+      // build update Obj, which needs to include the flavour record ids
+      const updateObj = { ...nrptiRecord, _id: existingRecord._id };
+
+      updateObj.updatedBy = (this.auth_payload && this.auth_payload.preferred_username) || '';
+      updateObj.dateUpdated = new Date();
+      updateObj.sourceDateUpdated = new Date();
```

Comment:
Should this not be a field populated from the datasource itself?

Reply:
Good catch!  That's actually not needed.  Removed in https://github.com/bcgov/NRPTI/pull/696/commits/a205ce08b7116d83ab23fb39ec28a1561c8e14f8